src/store/products/products.types.ts

```diff
@@ -0,0 +1,45 @@
+// Entities
+
+export type ProductId = number;
+
+export interface Product {
+  product_id: ProductId;
+  name: string;
+  purchase_cost: number;
+}
```

Comment:
is 'interface' a type of type so to speak. Here we define a type of object which must have it's first value a custom type defined above, it's second and third values are default types of number and string... 

But since 'ProductId' is defined to be of type 'number' what is the advantage of passing that in , rather than state 'product_id: ProductId'?

Reply:
So re `ProductId` vs `number` it's best practice to structure the types in this way in case something changes in future? In this case it's unlikely, but in other cases maybe it would be. Is is also just good practice to state contingencies explicitly? I.e. because ProductId is defined separately, because it will need accessed separately, then the interface should be enslaved to that definition, because ProductId is primary, and not the interface?